usecase/todo.go

```diff
@@ -0,0 +1,56 @@
+package usecase
+
+import (
+	"gorm.io/gorm"
+	"peanut/domain"
+	"peanut/repository"
+)
+
+type TodoUsecase interface {
+	CreateTodo(userId int, u domain.Todo) (err error)
+	ListTodo(userId int) (todos []domain.Todo, err error)
+	UpdateTodo(userId int, u domain.Todo) (err error)
+	DeleteTodo(userId int, todoId int) (err error)
+}
+
+type todoUsecase struct {
+	TodoRepo repository.TodoRepo
+}
+
+func NewTodoUsecase(db *gorm.DB) TodoUsecase {
+	return &todoUsecase{
+		TodoRepo: repository.NewTodoRepo(db),
+	}
+}
+
+func (uc *todoUsecase) CreateTodo(userId int, u domain.Todo) (err error) {
+	_, err = uc.TodoRepo.CreateTodo(u, uint(userId))
+	if err != nil {
+		return err
+	}
+	return
+}
+
+func (uc *todoUsecase) ListTodo(userId int) (todos []domain.Todo, err error) {
+	todos, err = uc.TodoRepo.ListTodo(userId)
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+func (uc *todoUsecase) UpdateTodo(userId int, u domain.Todo) (err error) {
```

Comment:
domain.Todo thì em nên đặt tên biến liên quan. a đặt là u vì trước đó a dùng cho domain User :))
```suggestion
func (uc *todoUsecase) UpdateTodo(userId int, td domain.Todo) (err error) {
```